internal/os/system/api.go

```diff
@@ -37,3 +50,51 @@ func (APIImplementor) GetBIOSSerialNumber() (string, error) {
 	}
 	return lines[1], nil
 }
+
+func (APIImplementor) GetService(name string) (*ServiceInfo, error) {
+	script := `Get-Service -Name $env:ServiceName | Select-Object DisplayName, Status, StartType | ` +
+		`ConvertTo-JSON`
+	cmd := exec.Command("powershell", "/c", script)
+	cmd.Env = append(os.Environ(), fmt.Sprintf("ServiceName=%s", name))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("error querying service name=%s. cmd: %s, output: %s, error: %v", name, cmd, string(out), err)
+	}
+
+	var serviceInfo ServiceInfo
+	err = json.Unmarshal(out, &serviceInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &serviceInfo, nil
+}
+
+func (APIImplementor) StartService(name string) error {
+	script := `Start-Service -Name $env:ServiceName`
+	cmd := exec.Command("powershell", "/c", script)
+	cmd.Env = append(os.Environ(), fmt.Sprintf("ServiceName=%s", name))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error starting service name=%s. cmd: %s, output: %s, error: %v", name, cmd, string(out), err)
+	}
+
+	return nil
+}
+
+func (APIImplementor) StopService(name string, force bool) error {
+	script := `Stop-Service -Name $env:ServiceName -Force:$([System.Convert]::ToBoolean($env:Force))`
```

Comment:
Same comment as before: Is there a reason to set the `servicename` and `force` parameters through environment variables here? Why not use sprintf directly rather than a couple of lines below?

Reply:
see above, tell me what you think